client/src/helpers/getDownloadOptionDetails.js

```diff
@@ -31,15 +31,26 @@ const formatFileItemByKey = (key, computedFile) => {
 export const getDownloadOptionDetails = (computedFile) => {
   const {
     metadata_only: metadataOnly,
+    portal_metadata_only: portalMetadataOnly,
     modality,
     project,
     sample
   } = computedFile
+  // helpers for info and resourceType
+  const getInfo = () => {
+    if (portalMetadataOnly) return metadataResourceInfo.ALL
+    if (metadataOnly) return metadataResourceInfo.PROJECT
+    return modalityResourceInfo[modalityResourceKey + suffix]
+  }
+  const getResourceType = () => {
+    if (portalMetadataOnly) return 'All'
+    return project ? 'Project' : 'Sample'
+  }
```

Comment:
To avoid nested ternary operators, I created these two helper methods to manage the correct return values for `info` and `resourceType`.